modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/P2PConnectionInfo.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.syntax.all._
+
+import com.comcast.ip4s.{Host, IpAddress, Port}
+import derevo.cats.{eqv, show}
+import derevo.derive
+import fs2.data.csv.CellDecoder
+
+@derive(eqv, show)
+case class P2PConnectionInfo(
+  host: Host,
+  p2pPort: Port
+)
+
+object P2PConnectionInfo {
+
+  implicit val maybeConnectionInfoCellDecoder: CellDecoder[Option[P2PConnectionInfo]] =
+    CellDecoder.stringDecoder.emap { value =>
+      decode(value).asRight
+    }
+
+  val delimiter = ":"
```

Comment:
Any reason why this isn't local to `decode` ?

Reply:
It is in the same object, if that's what you mean. I didn't put it in the same function because I didn't want to have to redeclare and initialize the variable every single time.

Reply:
Where do you need to redeclare it? Isn't it used once only?